capi/src/write/pages.ts

```diff
@@ -135,9 +142,20 @@ export async function pages(ctx: t.Ctx): Promise<void> {
       }
 
       await fs.ensureDir(outDir);
-      await fs.writeFile(pathDeduction.destinationPath, JSON.stringify(page), {
-        encoding: "utf8",
-      });
+      // hash file name to deal with aggressive adblockers
+      const paths = pathDeduction.destinationPath.split("/");
+      let fileName = paths.pop();
+      if (!fileName) {
+        fileName = "";
```

Comment:
In which cases will `fileName` be false-y?

Both `""` and `"/"` split & pop off `""`.

Reply:
This was for Typescript -- `Array.pop()` returns `string | undefined` where `undefined` is if the array is empty.  But it looks like you can just say `const fileName = paths.pop() as any` and it should compile, so thank you!

Reply:
Oooh, I was wondering why. This is was for type-safety!  Yes, I can see how `Array.pop()` could returned `undefined`, so no worries.

Since it's a false-y check, `|| ""` could also work.  Sorry for distracting!